src/Creator/InvoiceCreator.php

```diff
@@ -37,8 +38,9 @@ public function __construct(
 
     public function __invoke(string $orderNumber, \DateTimeInterface $dateTime): void
     {
-        /** @var OrderInterface $order */
-        $order = $this->orderRepository->findOneByNumber($orderNumber);
+        /** @var OrderInterface|null $order */
+        $order = $this->orderRepository->findOneBy(['number' => $orderNumber]);
```

Comment:
Why did you change that in this PR?

Reply:
Because `findOneByNumber` refers to OrderRepository method, which verify order is not on cart state. So we have a null value using repository method.
```php
    public function findOneByNumber(string $number): ?OrderInterface
    {
        return $this->createQueryBuilder('o')
            ->andWhere('o.state != :state')
            ->andWhere('o.number = :number')
            ->setParameter('state', OrderInterface::STATE_CART)
            ->setParameter('number', $number)
            ->getQuery()
            ->getOneOrNullResult()
        ;
    }
```

Reply:
So what is the reason the order is in cart state? Because in the moment of creating an invoice the checkout should be completed and the order shouldn't be in the cart state anymore. I guess, but I'm not sure, there is a difference with the moment of dispatching the event between state machine callback and the doctrine events.

Reply:
At this moment the order `state` is `new` instead of `cart` in the database.